prefect_gcp/cloud_storage.py

```diff
@@ -1140,3 +1214,71 @@ async def upload_from_folder(
         if num_uploaded == 0:
             self.logger.warning(f"No files were uploaded from {from_folder}.")
         return bucket_folder
+
+    @sync_compatible
+    async def upload_from_dataframe(
+        self,
+        df: DataFrame,
+        to_path: str,
+        output_format: Union[str, OutputFormat] = OutputFormat.CSV_GZIP,
```

Comment:
Thoughts on naming this `upload_format`?
```suggestion
        upload_format: Union[str, OutputFormat] = DataFrameUploadFormat.CSV_GZIP,
```

Reply:
@ahuang11 I feel like `DataFrameSerializationFormat` would be more accurate. Can we go with that?

Reply:
Sure